Add documentation on embedding views containing joins (#536)
PostgREST/postgrest-docs: docs/api.rst

  If view definitions change you must refresh PostgREST's schema cache for this to work properly. See the section :ref:`schema_reloading`.

Embedding Views containing Joins
--------------------------------

> The same embedding rules applies to views with joined tables, that is, the column of any table participating in the join with a **foreign key** definition must be present in the view.

Reviewer: On views that contain joins, PostgREST will infer the joined tables' foreign keys based on the source of the columns defined for the view.

For instance, let's create a view called ``film_competition_view`` that will join three tables:

.. code-block:: postgres

  CREATE VIEW film_competition_view AS
  SELECT
   films.title as film_title
   , competitions.name as competition_name

>    , film_id

Reviewer: Needs alignment
```sql
  SELECT
     films.title as film_title
   , competitions.name as competition_name
   , film_id
```

  FROM
   nominations
   JOIN films ON films.id = nominations.film_id

>    JOIN competitions ON competitions.id = nominations.competition_id;

Reviewer: It's subjective, but I read it better in this way:
```sql
  FROM nominations
  JOIN films ON films.id = nominations.film_id
  JOIN competitions ON competitions.id = nominations.competition_id;
```

The view column can embed the table ``films`` because it has the ``film_id`` column that has a foreign key defined.

> But what if you want to embed the view with the ``actors`` table? The ``film_id`` column alone will not allow it since it belongs to ``nominations`` and there is no relationship between that table and ``actors`` directly. In this case, you need to add the ``film.id`` column which does have the foreign key to the ``actors`` table.

Reviewer: > ``film.id`` column which does have the foreign key to the ``actors`` table.

The film doesn't have an FK to the `actors` table right? Only an m2m relationship through `roles`(which has the FK)

Author: True, I clarified the explanation now.

> But what if you want to embed the view with the ``actors`` table? The ``film_id`` column alone will not allow it since it belongs to ``nominations``, which doesn't have a relationship to ``actors``. In this case, you need to select the ``films.id`` column from the ``films`` table, which has a many-to-many relationship with ``actors``.

Reviewer: But what if you want to embed the view with the ``actors`` table? The ``film_id`` column alone will not allow it since it belongs to ``nominations``, which doesn't have a relationship to ``actors``. In this case, you need to select the ``id`` column from the ``films`` table, which has a many-to-many relationship with ``actors``.

.. code-block:: postgres

  CREATE VIEW film_competition_view AS
  SELECT
   films.title as film_title

>    , competitions.name as competition_name

Reviewer: ```sql
  SELECT
     films.title as film_title
   , competitions.name as competition_name
```

   -- To keep the relationship with the films table, this column is renamed
   , film_id as nominations_film_id
   -- Adding this column allows the relationships we want
   , films.id as film_id
  FROM
   nominations
   JOIN films ON films.id = nominations.film_id
   JOIN competitions ON competitions.id = nominations.competition_id;

> Now the view can embed both the ``films`` and ``actors`` tables:

Reviewer: Now the view can be embedded with both the ``films`` and ``actors`` tables:

  If view definitions change you must refresh PostgREST's schema cache for this to work properly. See the section :ref:`schema_reloading`.

Embedding Views Containing Joins
--------------------------------

The same embedding rules applies to views with joined tables, that is, the column of any table participating in the join with a **foreign key** definition must be present in the view.

For instance, let's create a view called ``film_competition_view`` that will join three tables:

.. code-block:: postgres

  CREATE VIEW film_competition_view AS
  SELECT
   films.title as film_title
   , competitions.name as competition_name
   , film_id
  FROM
   nominations
   JOIN films ON films.id = nominations.film_id
   JOIN competitions ON competitions.id = nominations.competition_id;

> The view can be embedded with the table ``films`` because it has the ``film_id`` column. But what if you want to embed it with the ``actors`` table, for instance? The ``film_id`` column alone will not allow it since it belongs to ``nominations`` and there is no relationship that can be inferred between that table and ``actors``. In this case, you need to select the ``film.id`` column which will allow any relationship that worked with the ``films`` table (using ``film.id`` as a foreign key) to also work for the view.

Reviewer: ```suggestion
The view can be embedded with the table ``films`` because it has the ``film_id`` column.

But what if you want to embed it with the ``actors`` table? The ``film_id`` column alone will not allow it since it belongs to ``nominations``, which doesn't have a relationship with ``actors``. In this case, you need to select the ``film.id`` column, that comes from ``films`` which has a many-to-many relationship with ``actors``.
```

Reviewer: ```suggestion
The view can be embedded with the table ``films`` because it has the ``film_id`` column, which comes from ``nominations`` and has a relationship with ``films``.

But what if you want to embed it with the ``actors`` table? The ``film_id`` column alone will not allow it since it belongs to ``nominations``, which doesn't have a relationship to``actors``. In this case, you need to select the ``film.id`` column, that comes from ``films`` and has a many-to-many relationship with ``actors``.
```
